Source/Xamarin/Prism.Forms/Common/PageUtilities.cs

```diff
@@ -47,6 +47,7 @@ private static void DestroyChildren(Page page)
         {
             if (page is MasterDetailPage)
             {
+                DestroyPage(((MasterDetailPage)page).Master);
```

Comment:
This is not about the actual MasterDetailPage, it's about destroying the MasterDetailPage.Master page.  I am destroying the Detail, but totally forgot about the Master.